src/context/WalletProvider/MobileWallet/components/MobileSuccess.tsx

```diff
@@ -33,23 +31,11 @@ export const MobileSuccess = ({ location }: MobileSetupProps) => {
 
         if (mnemonic) {
           await wallet.loadDevice({ mnemonic, deviceId })
-          const { name, icon } = MobileConfig
           const walletLabel = vault?.label ?? 'label'
-          dispatch({
-            type: WalletActions.SET_WALLET,
-            payload: {
-              wallet,
-              name,
-              icon,
-              deviceId,
-              meta: { label: walletLabel },
-              connectedType: KeyManager.Mobile,
-            },
-          })
-          dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+
           localWallet.setLocalWalletTypeAndDeviceId(KeyManager.Mobile, deviceId)
           localWallet.setLocalNativeWalletName(walletLabel)
-          appDispatch(setWelcomeModal({ show: true }))
```

Comment:
issue(blocking): revert me, or if that was intended, reply to this comment with the rationale for viz
 We're not setting welcome modal to true anymore, and AFAIK after grepping for `setWelcomeModal`, aren't setting it to true anywhere else in the codebase for mobile anymore.
We'be probably gone one too deep on the red side in this diff? i.e  `setWelcomeModal` is still in the deps on l.58 below which indicates this diff was unintended.

Reply:
Not valid anymore as discussed we prefer the quickfix regarding the complexity of `WalletProvider`, changed in [5eacb01](https://github.com/shapeshift/web/pull/7280/commits/5eacb015dbea0fa87977f66ebe16bb50b6338894)